test/test_common.hpp

```diff
@@ -629,11 +631,11 @@ namespace xt
 
         // shaped_xiterator
         {
-            std::vector<size_t> shape(rm.m_shape.size() + 1);
+            shape_type shape(rm.m_shape.size() + 1);
             std::copy(rm.m_shape.begin(), rm.m_shape.end(), shape.begin() + 1);
             shape[0] = 2;
-            auto iter = vec.xrbegin(shape);
-            auto iter_end = vec.xrend(shape);
+            auto iter = vec.template xrbegin<shape_type, layout_type::row_major>(shape);
+            auto iter_end = vec.template xrend<shape_type, layout_type::row_major>(shape);
```

Comment:
This shows that `layout_type` should probably be the first template parameter to `xbegin`, so that we don't need to specify `shape_type`.

When we make this change, which is backward incompatible, we will need to make a `0.11.0`